schema/schema_test.go

```diff
@@ -70,6 +70,37 @@ func TestSchemas_GetByID(t *testing.T) {
 	})
 }
 
+func TestSchemas_List(t *testing.T) {
+	ss := Schemas{
+		Schema{
+			ID: "foo",
+		},
+		Schema{
+			ID: "bar",
+		},
+		Schema{
+			ID: "foobar",
+		},
+		Schema{
+			ID: config.DefaultIdentityTraitsSchemaID,
+		},
+	}
+
+	t.Run("case=get all schema IDs", func(t *testing.T) {
+		ids_orig := []string{}
+		for _, s := range ss {
+			ids_orig = append(ids_orig, s.ID)
+		}
+		ids_list := []string{}
+		for _, s := range []Schema(ss.List(0, 4)) {
+			ids_list = append(ids_list, s.ID)
+		}
+		for _, id := range ids_orig {
+			require.Contains(t, ids_list, id)
+		}
+	})
```

Comment:
This test looks a bit weird to me, probably because you previously only returned the IDs?
What do you think about:

```suggestion
	t.Run("case=get all schemas", func(t *testing.T) {
		p0 := ss.List(0, 4)
		assert.Equal(t, ss, p0)
	})
	
	t.Run("case=smaller pages", func(t *testing.T) {
		p0, p1 := ss.List(0, 2), ss.List(2, 2)
		assert.Equal(t, ss, append(p0, p1...))
	}
	
	t.Run("case=indexes out of range", func(t *testing.T) {
		p0 := ss.List(-1, 10)
		assert.Equal(t, ss, p0)
	}
```

and maybe there is some other case I missed?